src/middlewared/middlewared/plugins/iscsi_/portal.py

```diff
@@ -64,7 +65,7 @@ async def config_extend(self, data, context):
         data['discovery_authgroup'] = data.pop('discoveryauthgroup')
         return data
 
-    @accepts()
+    @accepts(roles=['SHARING_ISCSI_PORTAL_READ'])
```

Comment:
This is not necessary, because, as the doc states https://github.com/truenas/middleware/blob/master/docs/source/middleware/roles.rst#id4, `SERVICE_READ` (and it will have access to calling `service.config`, `service.get_instance`, `service.query` and all  `service.*_choices` methods)

Reply:
Right.  Thanks!